articles/openstack/ostack-how-cloud-config.md

```diff
@@ -0,0 +1,66 @@
+---
+title: How to configure an OpenStack instance using cloud-config | UKCloud Ltd
+description: Describes how to use cloud-config (user-data) to configure an OpenStack instance
+services: openstack
+author: Steve Relf
+toc_rootlink: How To
+toc_sub1:
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Configure an instance using cloud-config
+toc_fullpath: How To/ostack-how-cloud-config.md
+toc_mdlink: ostack-how-cloud-config.md
+---
+
+# How to configure an OpenStack instance using cloud-config
+
+## Overview
+
+When you boot an instance in OpenStack you can pass configuration to the instance as cloud-config (sometimes called user-data). This enables you to quickly and easily make configuration changes at boot time.
+
+## Use cases
+
+The following list provides some examples of how you can use cloud-config to configure an instance:
+
+- Setting a default locale
+- Setting an instance hostname
+- Generating instance SSH private keys
+- Adding SSH keys to a user's `.ssh/authorized_keys` so they can log in
+- Setting up ephemeral mount points
+- Configuring network devices
+- Running config scripts at startup
+
+## Example
+
+You can deploy cloud-config via the command line or via the OpenStack Horizon GUI. In this guide, we use the CLI, but you can use the same script via the GUI by pasting it into the **Configuration** tab.
+
+As an example, let's boot an instance and run some arbitrary commands at boot time.
+
+1. Create a new text file called `user-data.txt`.
+
+2. Add the following to the new file.
+
+    ```
+    #cloud-config
```

Comment:
Is it worth emphasising that the #cloud-config "comment" is required? It caught me the first time, and has caught out a customer.